client/state/jetpack-connection-health/reducer.js

```diff
@@ -3,52 +3,93 @@ import {
 	JETPACK_CONNECTION_HEALTHY,
 	JETPACK_CONNECTION_MAYBE_UNHEALTHY,
 	JETPACK_CONNECTION_UNHEALTHY,
+	JETPACK_CONNECTION_HEALTH_REQUEST,
+	JETPACK_CONNECTION_HEALTH_REQUEST_FAILURE,
 } from 'calypso/state/action-types';
+import {
+	combineReducers,
+	keyedReducer,
+	withPersistence,
+	withSchemaValidation,
+} from 'calypso/state/utils';
+import { schema } from './schema';
 
-export const jetpackConnectionHealth = ( state = {}, action ) => {
+/**
+ * `Reducer` function which handles Jetpack connection health state changes
+ * @param {Object} state - current state
+ * @param {Object} action - action
+ * @returns {{ is_healthy?: boolean, error?: string, jetpack_connection_problem?: boolean }} updated state - Jetpack connection health
+ */
+export const connectionHealth = withPersistence( ( state = {}, action ) => {
 	switch ( action.type ) {
 		case JETPACK_CONNECTION_HEALTHY: {
-			const { siteId } = action;
-
 			return {
 				...state,
-				[ siteId ]: {
-					...state[ siteId ],
-					jetpack_connection_problem: false,
-					is_healthy: true,
-					error: '',
-				},
+				jetpack_connection_problem: false,
+				error: '',
 			};
 		}
 
 		case JETPACK_CONNECTION_MAYBE_UNHEALTHY: {
-			const { siteId } = action;
-
 			return {
 				...state,
-				[ siteId ]: {
-					...state[ siteId ],
-					jetpack_connection_problem: true,
-				},
+				jetpack_connection_problem: true,
 			};
 		}
 
 		case JETPACK_CONNECTION_UNHEALTHY: {
-			const { siteId, errorCode } = action;
+			const { errorCode } = action;
 
 			return {
 				...state,
-				[ siteId ]: {
-					...state[ siteId ],
-					jetpack_connection_problem: true,
-					is_healthy: false,
-					error: errorCode,
-				},
+				jetpack_connection_problem: true,
+				error: errorCode,
 			};
 		}
 	}
 
 	return state;
-};
+} );
+
+/**
+ * `Reducer` function which handles request errors concerning Jetpack connection health API call.
+ * @param {string} state - current state
+ * @param {Object} action - action
+ * @returns {string} updated state - request error code
+ */
+export const requestError = withPersistence( ( state = '', action ) => {
+	switch ( action.type ) {
+		case JETPACK_CONNECTION_HEALTH_REQUEST:
+			return '';
```

Comment:
I still don't understand why an empty string is better than `null` or `undefined` here.

Reply:
I switched to null. The reason was that at some point I had encountered an issue, where when the state was `null` it wasn't saved in the localStorage.
I tested it though and it seems that the functionality works with `null` so it didn't make sense to fallback to an empty string instead.